### PR TITLE
Simplify code to remove and put new entities.

### PR DESCRIPTION
**Pull request by a1exsh**

### CODE REVIEW

File: zmon-agent.py [hunk 1]
```diff
@@ -279,54 +279,38 @@ def main():
...
+                auth = None
+
             for e in to_remove:
                 print "removing instance: {}".format(e)
 
-                if os.getenv('zmon_user', None) is not None:
-                    r = requests.delete(args.entityservice+"{}/".format(e), auth=HTTPBasicAuth(os.getenv('zmon_user', None), os.getenv('zmon_password', None)))
-                else:
-                    r = requests.delete(args.entityservice+"{}/".format(e))
+                r = requests.delete(args.entityservice+"{}/".format(e), auth=auth)
```
Comment (hjacobs) [inline zmon-agent.py]: make it simpler: "auth" keyword accepts a simple tuple with (user, pass)

File: zmon-agent.py [hunk 2]
```diff
@@ -279,54 +279,38 @@ def main():
                 if not e["id"] in current_entities:
                     to_remove.append(e["id"])
 
+            if os.getenv('zmon_user', None) is not None:
```
Comment (hjacobs) [inline zmon-agent.py]: I dislike the repeated "None", also it should consider empty strings as "False".

Simply use:

```python
if os.getenv('zmon_user'):
   ...

**Comment by hjacobs** [comment]
Thanks for the pull request :smile:

File: zmon-agent.py (hunk 2 above)
Comment (a1exsh) [inline zmon-agent.py]: Good point.

**Comment by a1exsh** [comment]
Could you merge this please?  I'd like to add discovery of some new types of entities, but see no point doing that on old code.

**Pull request merged**